src/Razor/Microsoft.AspNetCore.Razor.Language/src/Components/ComponentTypeParamDirective.cs

```diff
@@ -7,45 +7,23 @@ namespace Microsoft.AspNetCore.Razor.Language.Components
 {
     internal class ComponentTypeParamDirective
```

Comment:
```suggestion
    internal static class ComponentTypeParamDirective
```

If you feel like it.

Reply:
This is a good recommendation. Do we have an analyzer for this?

I'll defer doing it in this PR and do a pass on this and other Directive types to make them all static just so we don't leave things in a state where the definitions are inconsistent.

Reply:
AFAIK no. This is probably the closest: https://github.com/dotnet/runtime/issues/49944